pkg/virtualization/core/virtualsystem/virtualmachine.go

```diff
@@ -836,6 +977,43 @@ func (vm *VirtualMachine) GetVirtualHardDiskByPath(path string) (vhd *disk.Virtu
 	return
 }
 
+func (vm *VirtualMachine) GetAttachedVirtualHardDisks() (vhdPaths []string, err error) {
+	col, err := vm.GetVirtualHardDisks()
+	if err != nil {
+		return
+	}
+	defer col.Close()
+
+	for _, inst := range col {
+		tmpvhd, err1 := disk.NewVirtualHardDisk(inst.WmiInstance)
+		if err1 != nil {
+			err = err1
+			return
+		}
+
+		vhdclone, err1 := tmpvhd.Clone()
+		if err1 != nil {
+			err = err1
+			return
+		}
+
+		retVhd, err1 := disk.NewVirtualHardDisk(vhdclone)
+		if err1 != nil {
+			err = err1
+			return
+		}
+
+		vhdpath, err1 := retVhd.GetPropertyHostResource()
```

Comment:
What if vhdpath has more than one element?

Reply:
Typically, the VHD resource will only contain 1 element as per documentation - https://learn.microsoft.com/en-us/previous-versions/cc136901(v=vs.85)
This resource is probably kept as array of strings because it's also used to represent other resources like processors. Let me add a check to ensure we get 1 property.